addons/udes_advanced_packaging/models/stock_picking.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+from collections import defaultdict
+
+from odoo import models, _
+from odoo.exceptions import ValidationError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _should_reserve_full_packages(self):
+        """Method to determine if picking should reserve entire packages"""
+        self.picking_type_id.ensure_one()
+        return self.picking_type_id.u_reserve_as_packages
+
+    def _reserve_full_packages(self):
+        """
+        If the picking type of the picking in self has full package
+        reservation enabled, whole packages will be reserved.
+        """
+        MoveLine = self.env["stock.move.line"]
+
+        # do not reserve full packages when bypass_reserve_full packages
+        # is set in the context as True
+        if not self.env.context.get("bypass_reserve_full_packages"):
+            pickings = self.filtered(lambda p: p._should_reserve_full_packages())
+            pickings.move_line_ids.package_id
```

Comment:
Seems that we are missing `.mapped()` here : `pickings.mapped("move_line_ids.package_id")`

Reply:
Forgot they changed this.